File: webapp/HIN_WEB/Models/DealRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace HIN_WEB.Models
{
    public class DealRequest
    {
        public DateTime? From { get; set; }
        public DateTime? To { get; set; }

        public List<Users> users { get; set; }
    }
    public class WinLossRequest : DealRequest
    {

    }
    public class WonDealsRequest : DealRequest
    {

    }
    public class DealsLostByReasonRequest : DealRequest
    {
        public List<Reason> reasons { get; set; }
    }
    public class RevenueforecastbyrepRequest : DealRequest
    {

    }
    public class LeadConversionRequest : DealRequest
    {

    }
    public class ProposalClosedByRepRequest : DealRequest
    {

    }
    public class OpportunitiesovertimeRepRequest : DealRequest
    {

    }
    public class OpportunitieslostbyreasonRequest : DealRequest
    {

    }
    public class ProposalsclosedbycustomerRequest : DealRequest
    {

    }



}

File: webapp/HIN_WEB/ClientApp/src/app/model/dealrequest.ts
import { Users } from "../users/users";
import { Reason } from "./reason";

export class Dealrequest {
  public From: Date;
  public To: Date;
  public users: Array<Users>;
  public reasons: Array<Reason>;

}
export class WinLossRequest {

  public WinLossRequest: Dealrequest = new Dealrequest;

}
export class WonDealsRequest {

  public WonDealsRequest: Dealrequest = new Dealrequest;

}
export class DealslostyyreasonRequest {

  public DealslostyyreasonRequest: Dealrequest = new Dealrequest;
 

}
export class RevenueforecastbyrepRequest {

  public RevenueforecastbyrepRequest: Dealrequest = new Dealrequest;
  
}
export class LeadConversionRequest {

  public LeadConversionRequest: Dealrequest = new Dealrequest;

}
export class ProposalclosedbyrepRequest {

  public ProposalclosedbyrepRequest: Dealrequest = new Dealrequest;

}
export class OpportunitiesovertimeRequest {

  public OpportunitiesovertimeRequest: Dealrequest = new Dealrequest;

}
export class OpportunitieslostbyreasonRequest {

  public OpportunitieslostbyreasonRequest: Dealrequest = new Dealrequest;

}
export class ProposalsclosedbycustomerRequest {

  public ProposalsclosedbycustomerRequest: Dealrequest = new Dealrequest;

}




